sfge/src/main/java/com/salesforce/rules/fls/apex/operations/StandardConditionDecomposer.java

```diff
@@ -0,0 +1,123 @@
+        if (children.size() > 1) {
+            throw new TodoException("Need to handle more than one child: " + children);
+        }
+        final BaseSFVertex child = children.get(0);
+        // StandardConditions are the contents of an IF clause's test. A Positive condition means
+        // that the path
+        // expects the expression to be satisfied, so we should limit our search to the clauses that
+        // need
+        // to be satisfied for this to be the case. Meanwhile, a Negative condition means that the
+        // path expects
+        // the expression to be unsatisfied, so we should limit our search to clauses that need to
+        // be unsatisfied
+        // for that to happen.
+        boolean seekSatisfiedClauses = vertex instanceof StandardConditionVertex.Positive;
+        return recursivelyDecomposeChildren(child, seekSatisfiedClauses);
+    }
+
+    private static List<BaseSFVertex> recursivelyDecomposeChildren(
+            BaseSFVertex vertex, boolean seekingSatisfied) {
+        // Create an empty list to which we'll add all the clauses that must be processed.
+        List<BaseSFVertex> results = new ArrayList<>();
+        // How we process a vertex depends on its type.
+        if (vertex instanceof BooleanExpressionVertex) {
+            BooleanExpressionVertex booleanVertex = (BooleanExpressionVertex) vertex;
+            results.addAll(decomposeBooleanExpression(booleanVertex, seekingSatisfied));
+        } else if (vertex instanceof PrefixExpressionVertex) {
+            // For a prefix expression, pull off the first child and recursively call on that.
+            // If it's a NOT-expression, we also switch whether we're seeking satisfied or
+            // unsatisfied clauses, because we're now expecting the opposite of whatever we
+            // were expecting previously.
+            PrefixExpressionVertex prefixVertex = (PrefixExpressionVertex) vertex;
+            BaseSFVertex child = vertex.getChild(0);
+            results.addAll(
+                    recursivelyDecomposeChildren(
+                            child,
+                            prefixVertex.isOperatorNegation()
+                                    ? !seekingSatisfied
+                                    : seekingSatisfied));
+        } else if (seekingSatisfied) {
+            // All other vertex types are base cases, so we don't recurse. Instead, if this vertex
+            // is one that
+            // we're expecting to be satisfied, we should add it to our list.
+            results.add(vertex);
+        }
+        return results;
+    }
+
+    private static List<BaseSFVertex> decomposeBooleanExpression(
+            BooleanExpressionVertex vertex, boolean seekingSatisfied) {
+        List<BaseSFVertex> results = new ArrayList<>();
+        BaseSFVertex lhs = vertex.getLhs();
+        BaseSFVertex rhs = vertex.getRhs();
+        boolean eitherSideTrue = True.isLiterallyTrue(lhs) || True.isLiterallyTrue(rhs);
```

Comment:
Would be helpful to have code snippets for each of these to visualize what the boolean is looking for.
Something like, `(value == true)` for `eitherSideTrue`.